axelrod/fingerprint.py

```diff
@@ -0,0 +1,294 @@
+            produce more coordinates that will be closer together.
+
+        Returns
+        ----------
+        edges : list of tuples
+            A list containing tuples of length 2. All tuples will have either 0
+            or 1 as the first element. The second element is the index of the
+            corresponding probe (+2 to allow for including the Strategy and it's
+            Dual).
+
+        tournament_players : list
+            A list containing instances of axelrod.Player. The first item is the
+            original player, the second is the dual, the rest are the probes.
+
+        """
+        probe_coordinates = create_coordinates(step)
+        self.coordinates = probe_coordinates
+        edges = create_edges(probe_coordinates)
+
+        dual = DualTransformer()(self.strategy)()
+        probe_players = create_probes(self.probe, probe_coordinates)
+        tournament_players = [self.strategy(), dual] + probe_players
+
+        return edges, tournament_players
+
+    def fingerprint(self, turns=50, repetitions=10, step=0.01, processes=None,
+                    filename=None):
+        """Build and play the spatial tournament.
+
+        Creates the probes and their edges then builds a spatial tournament
+        where the original strategy only plays probes whose coordinates sum to
+        less than 1 (or equal). Probes whose coordinates sum to more than 1 play
+        the Dual Strategy.
+
+        Parameters
+        ----------
+        turns : integer, optional
+            The number of turns per match
+        repetitions : integer, optional
+            The number of times the round robin should be repeated
+        step : float, optional
+            The separation between each coordinate. Smaller steps will
+            produce more coordinates that will be closer together.
+        processes : integer, optional
+            The number of processes to be used for parallel processing
+        """
+        in_memory = False
+        if on_windows and (filename is None):
+            in_memory = True
+        elif filename:
+            outputfile = open(filename, 'w')
+        else:
+            outputfile = NamedTemporaryFile(mode='w')
+        filename = outputfile.name
+
+        edges, tourn_players = self.construct_tournament_elements(step)
+        self.step = step
+        self.spatial_tourn = axl.SpatialTournament(tourn_players,
+                                                   turns=turns,
+                                                   repetitions=repetitions,
+                                                   edges=edges)
+        self.spatial_tourn.play(build_results=False,
+                                filename=filename,
+                                processes=processes,
+                                in_memory=in_memory)
+        self.interactions = read_interactions(filename)
+        self.data = generate_data(self.interactions, self.coordinates, edges)
```

Comment:
As well as setting `self.data` we could also return the data. This would be akin to all the `{match, tournament, moran}.play()` methods which not only set something but also return it...